src/SFML/Audio/SoundFileReaderFlac.cpp

```diff
@@ -172,6 +172,69 @@ void streamMetadata(const FLAC__StreamDecoder*, const FLAC__StreamMetadata* meta
         data->info.sampleCount  = meta->data.stream_info.total_samples * meta->data.stream_info.channels;
         data->info.sampleRate   = meta->data.stream_info.sample_rate;
         data->info.channelCount = meta->data.stream_info.channels;
+
+        // For FLAC channel mapping refer to: https://xiph.org/flac/format.html#frame_header
+        switch (data->info.channelCount)
+        {
+            case 0:
+                sf::err() << "No channels in FLAC file" << std::endl;
+                break;
+            case 1:
+                data->info.channelMap = {sf::SoundChannel::Mono};
+                break;
+            case 2:
+                data->info.channelMap = {sf::SoundChannel::FrontLeft, sf::SoundChannel::FrontRight};
+                break;
+            case 3:
+                data->info.channelMap = {sf::SoundChannel::FrontLeft,
+                                         sf::SoundChannel::FrontRight,
+                                         sf::SoundChannel::FrontCenter};
+                break;
+            case 4:
+                data->info.channelMap = {sf::SoundChannel::FrontLeft,
+                                         sf::SoundChannel::FrontRight,
+                                         sf::SoundChannel::BackLeft,
+                                         sf::SoundChannel::BackRight};
+                break;
+            case 5:
+                data->info.channelMap = {sf::SoundChannel::FrontLeft,
+                                         sf::SoundChannel::FrontRight,
+                                         sf::SoundChannel::FrontCenter,
+                                         sf::SoundChannel::BackLeft,
+                                         sf::SoundChannel::BackRight};
+                break;
+            case 6:
+                data->info.channelMap = {sf::SoundChannel::FrontLeft,
+                                         sf::SoundChannel::FrontRight,
+                                         sf::SoundChannel::FrontCenter,
+                                         sf::SoundChannel::LowFrequencyEffects,
+                                         sf::SoundChannel::BackLeft,
+                                         sf::SoundChannel::BackRight};
```

Comment:
I am assuming that the order of these channels matters, and that's why these three huge switches cannot be refactored to a separate function. Is that right?

Reply:
Yes, FLAC specifies channel ordering based on the number of channels.

https://xiph.org/flac/format.html#frame_header